### PR TITLE
CLOUDP-246693: Add ubuntu 2404

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

CLOUDP-246693
## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: build/ci/repo_config.yaml [hunk 1]
```diff
@@ -325,3 +325,17 @@ repos:
       - arm64
     repos:
       - apt/ubuntu/dists/jammy/mongodb-enterprise
+
+  - name: ubuntu2404
+    type: deb
+    code_name: "noble"
+    edition: enterprise
```
Comment (fmenezes) [inline build/ci/repo_config.yaml:332]: how about `org`?

Comment (blva) [reply build/ci/repo_config.yaml:360]: thanks, added~

File: tools/genevergreen/generate/generate.go [hunk 2]
```diff
@@ -59,12 +61,13 @@ var (
 		"centos8":         "rhel80",
 		"rhel9":           "rhel90",
 		"amazonlinux2":    "amazon2",
-		"amazonlinux2023": "amazon2023",
```
Comment (gssbzn) [inline tools/genevergreen/generate/generate.go:62]: intentional?

Comment (gssbzn) [reply tools/genevergreen/generate/generate.go:62]: oh moved bellow

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**